packages/dds/tree/src/test/simple-tree/schemaTypes.spec.ts

```diff
@@ -524,4 +546,111 @@ describe("schemaTypes", () => {
 		); // Different custom metadata
 		check(sf.identifier, sf.optional(sf.string), false); // Identifier vs. optional string
 	});
+
+	describe("Generic Schema", () => {
+		// Many of these cases should compile, but don't.
+		// This is likely due to `[FieldSchema<Kind, T>] extends [ImplicitFieldSchema] ? TrueCase : FalseCase` not getting reduced to `TrueCase`.
+		// This could be due to the compiler limitation noted in https://github.com/microsoft/TypeScript/issues/52144#issuecomment-2686250788
+
+		it("Generic construct", () => {
+			const sf = new SchemaFactory("test");
+
+			function makeInstanceImplicit<T extends ImplicitAllowedTypes>(
```

Comment:
I think docs for these functions would be useful

Reply:
I have added a bunch of docs